src/components/InputImage.vue

```diff
@@ -1,74 +1,128 @@
+      </div>
+    <input
+      type="file"
+      ref="image"
+      name="image"
+      @change="onFileChange($event.target.name, $event.target.files, validate)"
+      class="d-none">
+    <div class="mb-4"></div>
+    <transition-group tag="div" name="fade-transition">
+      <div v-for="(error, index) in validationErrors" :key="error">
+          <v-fade-transition>
+            <v-alert
+              :dense="alertDense"
+              :colored-border="alertColoredBorder"
+              :elevation="alertElevation"
+              :border="alertBorderLocation"
+              :type="alertType"
+              dismissible
+              @input="deleteError(index)"
+              >
+              {{error}}
+            </v-alert>
+          </v-fade-transition>
+      </div>
+    </transition-group>
+  </ValidationProvider>
 </template>
 
 <script>
+import { ValidationProvider } from 'vee-validate'
+import Camera from 'vue-material-design-icons/Camera'
+
 export default {
   name: 'InputImage',
+  components: {
+    ValidationProvider,
+    Camera
+  },
+  props: {
+    name: {
+      type: String,
+      required: true
+    },
+    icon: {
+      type: String
+    },
+    value: {
+      required: true
+    },
+    validationRules: {
+      type: Object,
+      required: true
+    },
+    alertType: {
+      type: String
+    },
+    alertElevation: {
+      type: String
+    },
+    alertDense: {
+      type: Boolean
+    },
+    alertBorderLocation: {
+      type: String
+    },
+    alertColoredBorder: {
+      type: Boolean
+    }
+  },
   data () {
     return {
-      errorDialog: null,
-      errorText: '',
-      uploadFieldName: 'image',
-      maxSize: 1024
+      validationErrors: [],
+      imageUrl: null
     }
   },
-  props: {
-    value: Object
-  },
   methods: {
     launchFilePicker () {
       this.$refs.image.click()
     },
-    onFileChange (fieldName, file) {
-      const { maxSize } = this
-      let imageFile = file[0]
-      if (file.length > 0) {
-        let size = imageFile.size / maxSize / maxSize
-        if (!imageFile.type.match('image.*')) {
-          // check whether the upload is an image
-          this.errorDialog = true
-          this.errorText = 'Please choose an image file'
-        } else if (size > 1) {
-          // check whether the size is greater than the size limit
-          this.errorDialog = true
-          this.errorText = 'Your file is too big! Please select an image under 1MB'
-        } else {
-          // Append file into FormData and turn file into image URL
-          let formData = new FormData()
-          let imageURL = URL.createObjectURL(imageFile)
-          formData.append(fieldName, imageFile)
-          // Emit the FormData and image URL to the parent component
-          this.$emit('input', { formData, imageURL })
-        }
+    deleteError (index) {
+      this.$delete(this.validationErrors, index)
+    },
+    async onFileChange (fieldName, file, validate) {
```

Comment:
`@change="onFileChange($event.target.name, $event.target.files, validate)"`

file or files???

Reply:
I think instead of passing files i will pass files[0], because user can only select one image, since multiply attribute is not present on image input.

Reply:
Addressed in 9356bcd43381fab684f6e47cdefc6d25aaad99f4

Reply:
Correct